src/span_context.js

```diff
@@ -0,0 +1,170 @@
+
+        return this._baggageHeaderCache;
+    }
+
+    /**
+     * Returns a normalize key.
+     *
+     * @param {string} key - The key to be normalized for a particular baggage value.
+     * @return {string} - The normalized key (lower cased and underscores replaced, along with dashes.)
+     **/
+    // todo(oibe) moved to span in newest opentracing version
+    _normalizeBaggageKey(key: string) {
+        let baggageHeaderCache = this.constructor.getBaggageHeaderCache();
+        if (key in baggageHeaderCache) {
+            return baggageHeaderCache[key];
+        }
+
+        // This stackoverflow claims this approach is slower than regex
+        // http://stackoverflow.com/questions/1144783/replacing-all-occurrences-of-a-string-in-javascript
+        let normalizedKey: string = key.toLowerCase().split('_').join('-');
+
+        if (Object.keys(baggageHeaderCache).length < 100) {
+            baggageHeaderCache[key] = normalizedKey;
+        }
+
+        return normalizedKey;
+    }
+
+    /**
+     * Sets a baggage value with an associated key.
+     *
+     * @param {string} key - The baggage key.
+     * @param {string} value - The baggage value.
+     **/
+    // todo(oibe) moved to span in newest opentracing version
+    setBaggageItem(key: string, value: string): void {
+        let normalizedKey = this._normalizeBaggageKey(key);
+        this._baggage[normalizedKey] = value;
+    }
+
+    /**
+     * Gets a baggage value with an associated key.
+     *
+     * @param {string} key - The baggage key.
+     * @return {string} value - The baggage value.
+     **/
+    // todo(oibe) moved to span in newest opentracing version
+    getBaggageItem(key: string): string {
+        let normalizedKey = this._normalizeBaggageKey(key);
+        return this._baggage[normalizedKey];
+    }
+
+    /**
+     * @return {boolean} - returns whether or not this span context was sampled.
+     **/
+    IsSampled() {
+        return this.flags & constants.SAMPLED_MASK === constants.SAMPLED_MASK;
+    }
+
+    /**
+     * @return {string} - returns a string version of this span context.
+     **/
+    toString() {
+        var parentId = this._parentId ? this._parentId.toString('hex') : '0';
+
+        return [
+            Utils.removeLeadingZeros(this._traceId.toString('hex')),
+            Utils.removeLeadingZeros(this._spanId.toString('hex')),
+            Utils.removeLeadingZeros(parentId),
+            this._flags.toString(16)
+        ].join(':');
+    }
+
+    /**
+     * @param {string} serializedString - a serialized span context.
+     * @return {SpanContext} - returns a span context represented by the serializedString.
+     **/
+    static fromString(serializedString: string): any {
+        var bufferValueOne: any = Utils.encodeInt64(1);
+
+        let headers:  any = serializedString.split(':');
+        if (headers.length !== 4) {
+            return null;
+        }
+
+        let traceId: any = Utils.encodeInt64(headers[0]);
```

Comment:
can this method throw exceptions? let's document, we'll need to revisit in the tracer.extract()